src/Marten/Events/Daemon/Coordination/SingleTenantProjectionDistributor.cs

```diff
@@ -0,0 +1,86 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using JasperFx.Core;
+using Marten.Storage;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+
+namespace Marten.Events.Daemon.Coordination;
+
+public class SingleTenantProjectionDistributor : IProjectionDistributor
+{
+    private readonly DocumentStore _store;
+    private readonly Cache<IMartenDatabase, AdvisoryLock> _locks;
+
+    public SingleTenantProjectionDistributor(DocumentStore store)
+    {
+        _store = store;
+
+        var logger = _store.Options.LogFactory?.CreateLogger<AdvisoryLock>() ??
+                     _store.Options.DotNetLogger ?? NullLogger<AdvisoryLock>.Instance;
+
+        _locks = new(db => new AdvisoryLock(db, logger));
+    }
+
+    public ValueTask<IReadOnlyList<IProjectionSet>> BuildDistributionAsync()
+    {
+        var database = _store.Storage.Database;
+        var projectionShards = _store.Options.Projections.AllShards();
+
+        var random = new Random();
```

Comment:
**SUGGESTION:** Same suggestion as @gfoidl about `Random.Shared`

Reply:
Good catch. Just missed it before.